pulpcore/pulpcore/app/viewsets/repository.py

```diff
@@ -178,6 +187,63 @@ def destroy(self, request, repository_pk, number):
         )
         return OperationPostponedResponse([async_result], request)
 
+    def validate_content_url(self, url):
+        """
+        Validates that the path belongs to a Content resource
+
+        Args:
+            url (str): href that needs validation
+
+        Returns:
+            PK of the Content
+
+        Raises:
+            :class:`rest_framework.exceptions.ValidationError`: When the URL is not a valid
+                Content resource
+        """
+
+        try:
+            resolver = resolve(urlparse(url).path)
```

Comment:
I'm surprised this is necessary. It seems like DRF should be able to get_or_404 from an href.